### PR TITLE
Nvim parity for queries

### PR DESCRIPTION
**Pull request by xcb-xwii**
Supersedes #50

- Correctly implement predicates to handle quantified captures
- Add unimplemented predicates
- Add negated predicates
- Dummy `set!` (for now)

### CODE REVIEW

File: highlights.lua [hunk 1]
```diff
@@ -4,9 +4,164 @@ local ts = require 'libraries.tree_sitter'
...
+		local startPt = n:start_point()
+		local endPt   = n:end_point()
+		local startRow, startCol = startPt:row() + 1, startPt:column() + 1
+		local endRow, endCol     = endPt:row() + 1, endPt:column()
+
+		return doc:get_text(startRow, startCol, endRow, endCol)
+	end
+
+	local function coerceToStr(n)
+		if type(b) ~= 'string' then
```
Comment (TorchedSammy) [inline highlights.lua:18]: `b` is not defined

**Review by TorchedSammy — CHANGES_REQUESTED** [review]

File: highlights.lua [hunk 2]
```diff
@@ -4,9 +4,164 @@ local ts = require 'libraries.tree_sitter'
...
+		['eq?'] = function(ns, m)
+			local str = coerceToStr(m)
+
+			for _, n in ipairs(ns:nodes()) do
+				if getSource(n) ~= str then return false end
+			end
+			return true
+		end,
+
+		['any-eq?'] = function(ns, m)
```
Comment (TorchedSammy) [inline highlights.lua:35]: this is a real predicate? used by real nvim-treesitter queries? isnt it just an inverse of eq? am i stupid or is this stupid?

Comment (xcb-xwii) [reply highlights.lua:35]: When the capture consists of multiple nodes, `pred?` checks if each of the individual nodes satisfies a condition. `any-pred?` checks if any one of the nodes satisfies the condition.

Comment (xcb-xwii) [reply highlights.lua:35]: See https://neovim.io/doc/user/treesitter.html#lua-treesitter-all-predicate

**Review by TorchedSammy — APPROVED** [review]

**Pull request merged**